scope_test.go

```diff
@@ -529,6 +575,36 @@ func TestTaggedSubScope(t *testing.T) {
 	}, r.histograms["foo.bar"].tags)
 }
 
+func TestTaggedSanitisedSubScope(t *testing.T) {
+	r := newTestStatsReporter()
+
+	ts := map[string]string{"env": "test:env"}
+	root, closer := NewRootScope(ScopeOptions{
+		Prefix:          "foo",
+		Tags:            ts,
+		Reporter:        r,
+		SanitiseOptions: &alphanumericSanitiserOpts,
+	}, 0)
+	defer closer.Close()
+
+	s := root.(*scope)
+
+	tscope := root.Tagged(map[string]string{"service": "test.service"}).(*scope)
+
+	r.cg.Add(1)
+	tscope.Counter("beep").Inc(1)
+
+	s.report(r)
+	tscope.report(r)
+	r.cg.Wait()
+
+	assert.EqualValues(t, 1, r.counters["foo.beep"].val)
```

Comment:
Wanted feedback on this behaviour – the implementation sanitises at the time of metric/scope construction. i.e. when we call `SubScope(_)`, or `Tagged()`, or `Counter(...)`. This means prefix concatenation can introduce chars invalid as defined by the sanitiser.

This example, we're specifying a prefix of `foo`, and permit `[a-zA-Z0-9-_]` as valid chars. When we create counter `foo`, we still end up with a counter named `foo.beep`. 

I prefer this behaviour to sanitising post the `fullyQualifiedName` call but I can see how it might be confusing to users. What do you think?